pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -956,13 +956,18 @@ public CompletableFuture<Optional<Topic>> getTopic(final String topic, boolean c
 
     public CompletableFuture<Optional<Topic>> getTopic(final String topic, boolean createIfMissing,
                                                        Map<String, String> properties) {
+        return getTopic(TopicName.get(topic), createIfMissing, properties);
+    }
+
+    public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, boolean createIfMissing,
```

Comment:
@nodece may you talk a bit detail how this change fixes the issue? And if a similar issue is in the master branch.

Reply:
See https://github.com/apache/pulsar/pull/17416